### PR TITLE
Build: compile detection was using a non-conforming package, causing cache to miss always.

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: compile doesn't properly check

## What I did

I made it check a different package, which is not abnormal

### CODE REVIEW

File: scripts/tasks/compile.ts [hunk 1]
```diff
@@ -19,7 +19,10 @@ export const compile: Task = {
       // To check if the code has been compiled as we need, we check the compiled output of
       // `@storybook/preview`. To check if it has been built for publishing (i.e. `--no-link`),
       // we check if it built types or references source files directly.
-      const contents = await readFile(resolve(codeDir, './lib/preview/dist/runtime.d.ts'), 'utf8');
```
Comment (ndelangen) [inline scripts/tasks/compile.ts:22]: To explain:

This is checking if the source of `'./lib/preview/dist/runtime.d.ts'` is this:
```
export * from '../src/index'
```

But of course it isn't! It's going to match the filename:

```
export * from '../src/runtime'
```

File: scripts/prepare/bundle.ts [hunk 2]
```diff
@@ -134,11 +138,19 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
...
+}: {
+  formats: Formats[];
+  entries: string[];
+  optimized: boolean;
+}) {
   const tsConfigPath = join(cwd, 'tsconfig.json');
   const tsConfigExists = await fs.pathExists(tsConfigPath);
 
-  const dtsBuild = formats[0] && tsConfigExists ? formats[0] : undefined;
+  const dtsBuild = optimized && formats[0] && tsConfigExists ? formats[0] : undefined;
```
Comment (ndelangen) [inline scripts/prepare/bundle.ts:153]: turned out we were always generating the definitions :(
oops.

File: scripts/tasks/compile.ts (hunk 1 above)
Comment (ndelangen) [reply scripts/tasks/compile.ts:22]: Also turns out I added a `;` at the end a while ago in the bundle script that I never added here...

File: scripts/prepare/bundle.ts (hunk 2 above)
Comment (ndelangen) [reply scripts/prepare/bundle.ts:153]: This might make local builds a bit faster again. Though this change hasn't been around for long.

**Comment by ndelangen** [comment]
In my few experiments `ts-node` seems just ever so slightly faster with the `--swc` flag turned on.

**Comment by ndelangen** [comment]
so `-S` is required in unix, but forbidden in `windows`... got it. great stuff

**Comment by ndelangen** [comment]
@valentinpalkovic as a windows users (or former windows user?) would you have any advice here on how to deal with this?

**Comment by ndelangen** [comment]
self-merging @tmeasday @shilman

**Comment by valentinpalkovic** [comment]
Unfortunately no :/ I havenʼt stumbled upon this issue so far

**Pull request merged**